client/boot/index.js

```diff
@@ -51,6 +52,9 @@ function init() {
 
 	debug( 'Starting Calypso. Let\'s do this.' );
 
+	// prune sync-handler records
+	syncHandler.pruneRecordsFrom( '30 minutes ago' );
```

Comment:
Also, that method doesn't accept a parameter. https://github.com/Automattic/wp-calypso/pull/3327/files#diff-fbc7b9c3d1ab50f852267129fe10abfaR222